File: C++/777. Swap Adjacent in LR String.cpp
class Solution {
public:
    bool canTransform(string s, string t) {
        int i=0,j=0;
        while(i<s.size()||j<s.size()){
            while(i<s.size()&&s[i]=='X')
            i++;
            while(j<s.size()&&t[j]=='X')
            j++;
            if(i==s.size()||j==s.size())
            return i==s.size()&&j==s.size();
            if((s[i]!=t[j])||(s[i]=='L'&&i<j)||(s[i]=='R'&&i>j))
            return false;
            i++;
            j++;
        }
        while(i<s.size()&&s[i]=='X')
        i++;
        while(j<s.size()&&t[j]=='X')
        j++;
        return i==s.size()&&j==s.size();
    }
};

File: JAVA/777. Swap Adjacent in LR String.java
class Solution {
    public boolean canTransform(String s, String t) {
        int i=0,j=0;
        while(i<s.length()&&j<t.length()){
            while(i<s.length()&&s.charAt(i)=='X')
            i++;
            while(j<t.length()&&t.charAt(j)=='X')
            j++;
            if(i==s.length()||j==s.length())
            return i==s.length()&&j==s.length();
            if((s.charAt(i)!=t.charAt(j))||(s.charAt(i)=='L'&&i<j)||(s.charAt(i)=='R'&& i>j))
            return false;
            i++;
            j++;
        }
        while (i < s.length() && s.charAt(i) == 'X') i++;
        while (j < t.length() && t.charAt(j) == 'X') j++;
        return i == s.length() && j == t.length();
    }
}